pkg/auth/authorizer/interfaces.go

```diff
@@ -41,6 +41,10 @@ type Attributes interface {
 
 	// The kind of object, if a request is for a REST object.
 	GetResource() string
+
+	// In case a special endpoint like /api, /healthz has been called. This can
+	// only yield a value when both GetNamespace() & GetResource() return none.
+	GetNonResourcePath() string
```

Comment:
@deads2k thoughts on the name? My desire for a nice name makes we want `GetPath()` (with documentation that it is only set for non-API requests), but my desire for complete clarity makes me want `GetNonResourcePath()`